Menus/TitleScreen.cpp

```diff
@@ -0,0 +1,608 @@
+
+		int starSmallBitmapCount = 4;
+		std::vector<BITMAP *> starSmallBitmaps = ConvertDoublePointerToVectorOfPointers(ContentFile("Base.rte/GUIs/Title/Stars/StarSmall.png").GetAsAnimation(starSmallBitmapCount), starSmallBitmapCount);
+
+		int starLargeBitmapCount = 1;
+		std::vector<BITMAP *> starLargeBitmaps = ConvertDoublePointerToVectorOfPointers(ContentFile("Base.rte/GUIs/Title/Stars/StarLarge.png").GetAsAnimation(starLargeBitmapCount), starLargeBitmapCount);
+
+		int starHugeBitmapCount = 2;
+		std::vector<BITMAP *> starHugeBitmaps = ConvertDoublePointerToVectorOfPointers(ContentFile("Base.rte/GUIs/Title/Stars/StarHuge.png").GetAsAnimation(starHugeBitmapCount), starHugeBitmapCount);
+
+		int starCount = (g_FrameMan.GetResX() * m_Nebula.GetBitmap()->h) / 1000;
+		for (int i = 0; i < starCount; ++i) {
+			Star newStar;
+			if (RandomNum() < 0.95F) {
+				newStar.Size = Star::StarSize::StarSmall;
+				newStar.Bitmap = starSmallBitmaps.at(RandomNum(0, starSmallBitmapCount - 1));
+				newStar.Intensity = RandomNum(0, 92);
+			} else if (RandomNum() < 0.85F) {
+				newStar.Size = Star::StarSize::StarLarge;
+				newStar.Bitmap = starLargeBitmaps.at(RandomNum(0, starLargeBitmapCount - 1));
+				newStar.Intensity = RandomNum(111, 185);
+			} else {
+				newStar.Size = Star::StarSize::StarHuge;
+				newStar.Bitmap = starHugeBitmaps.at(RandomNum(0, starLargeBitmapCount - 1));
+				newStar.Intensity = RandomNum(166, 185);
+			}
+			newStar.Position = Vector(RandomNum(0.0F, static_cast<float>(g_FrameMan.GetResX())), RandomNum(-100.0F, static_cast<float>(m_Nebula.GetBitmap()->h)));
+
+			m_BackdropStars.emplace_back(newStar);
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::CreateIntroSequenceSlides() {
+		std::string highRes = (g_FrameMan.GetResY() >= 680) ? "HD" : "";
+		for (int slideNum = 0; slideNum < m_IntroSlides.size(); ++slideNum) {
+			m_IntroSlides.at(slideNum) = ContentFile(("Base.rte/GUIs/Title/Intro/IntroSlide" + std::to_string(slideNum + 1) + highRes + ".png").c_str()).GetAsBitmap(COLORCONV_NONE, false);
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::Destroy() const {
+		for (BITMAP *slide : m_IntroSlides) {
+			if (slide) { destroy_bitmap(slide); }
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void TitleScreen::Update() {
+		if (m_SectionSwitch) { m_SectionTimer.Reset(); }
+		m_SectionElapsedTime = static_cast<float>(m_SectionTimer.GetElapsedRealTimeS());
+		m_SectionProgress = std::min((m_SectionDuration > 0) ? m_SectionElapsedTime / m_SectionDuration : 0, 0.9999F);
+
+		if (m_StationOrbitProgress >= 0.999F) { m_StationOrbitTimer.Reset(); }
```

Comment:
I assume it's arbitrary that you're checking against `0.999F` here and setting it to `0.9999F` above and below, right?

Reply:
There appears to be a hiccup when the station goes from 359 to 0 deg if checking for 0.9999 here.

Reply:
Hahaha wtf. Okay, so be it lol